### ISSUE TITLE
FAISS.add_embeddings is typed to take iterables but does not. 

### ISSUE THREAD

**Issue by startakovsky** [langchain-ai/langchain#5336]
### System Info

MacOS
Langchain Version 0.0.181
Python Version 3.11.3

### Who can help?

@eyurtsev I wasn't sure who to reach out to. The following is the signature for adding embeddings to FAISS: 

```python
FAISS.add_embeddings(
    self,
    text_embeddings: 'Iterable[Tuple[str, List[float]]]',
    metadatas: 'Optional[List[dict]]' = None,
    **kwargs: 'Any',
) -> 'List[str]'
```

Notice that `text_embeddings` takes an iterable. However, when I do this I get a failure with my iterable, but when wrapped in a `list` function then it is successful.

 

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```python
from langchain.vectorstores import FAISS
from langchain.embeddings import OpenAIEmbeddings

vs = FAISS.from_texts(['a'], embedding=OpenAIEmbeddings())
vector = OpenAIEmbeddings().embed_query('b')

# error happens with this next line, see "Expected behavior" below.
vs.add_embeddings(iter([('b', vector)]))

# no error happens when wrapped in a list
vs.add_embeddings(list(iter([('b', vector)])))
```

### Expected behavior

```bash
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
...
File ~/.pyenv/versions/3.11.3/envs/myenv/lib/python3.11/site-packages/faiss/class_wrappers.py:227, in handle_Index.<locals>.replacement_add(self, x)
    214 def replacement_add(self, x):
    215     """Adds vectors to the index.
    216     The index must be trained before vectors can be added to it.
    217     The vectors are implicitly numbered in sequence. When `n` vectors are
   (...)
    224         `dtype` must be float32.
    225     """
--> 227     n, d = x.shape
    228     assert d == self.d
    229     x = np.ascontiguousarray(x, dtype='float32')

ValueError: not enough values to unpack (expected 2, got 1)
```

**Comment by ninjapenguin** [comment]
This is fixed in the linked PR (#5367)

thanks for the reproducing case! 🙇

**Comment by ajeeto** [comment]
Hello,  I have a question 

I have created embedding for a csv file using open ai embeeding model

loader = CSVLoader('test1.csv', encoding="utf-8",csv_args={'delimiter': ','})
pages = loader.load_and_split()
len(pages)
embeddings = OpenAIEmbeddings(model=OPENAI_EMBEDDING_DEPLOYMENT_NAME,chunk_size=1)
db = FAISS.from_documents(documents=pages, embedding=embeddings)
db.save_local("./csvdata")
all good till here,
what i do not know is how can I embed another file say  test2.csv  and then  append the embedding to same vector store,
and save in same folder. any help will be greatly appreciated
